sentry-spring/src/main/java/io/sentry/spring/webflux/TransactionNameProvider.java

```diff
@@ -0,0 +1,30 @@
+package io.sentry.spring.webflux;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.web.reactive.HandlerMapping;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.util.pattern.PathPattern;
+
+/**
+ * Resolves transaction name using {@link ServerWebExchange#getRequest()} ()} and templated route
+ * that handled the request. To return correct transaction name, it must be used after request is
+ * processed by {@link
+ * org.springframework.web.reactive.result.method.RequestMappingInfoHandlerMapping} where {@link
+ * HandlerMapping#BEST_MATCHING_PATTERN_ATTRIBUTE} is set.
+ */
+@ApiStatus.Internal
+public final class TransactionNameProvider {
```

Comment:
Couldn't this just be a static thing?
Code using it doesn't seem to be mocking it

Reply:
Yep, can be static. Changed.